pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1597,6 +1602,21 @@ async def run(self):
                             if incorrect_changelogs or incorrect_modular_relation:
                                 log.debug("Updated data for package {}".format(result.nevra))
                                 await sync_to_async(result.save)()
+                            seen = []
```

Comment:
This all needs to go before the save() or else nothing will happen.  But also, you don't need to repeat the deduplication operations because they have already happened on the incoming package, so all you need to do is check if incoming and existing are different.

IMO, just checking the lengths is sufficient, it's not necessary to compare the exact paths again